### ISSUE TITLE
Cannot get to function correctly on WebThings gateway

### ISSUE THREAD

**Issue by da-wolff** [mozilla-iot/speed-test-adapter#6]
This is the log results on my system: 

2020-08-18 17:27:14.647 INFO   : Loading add-on: speed-test-adapter
2020-08-18 17:27:15.116 INFO   : getValue for property download for: Internet speed test returning 0

I can load the webpage https://fast.com on my raspberry pi but this add-on does not work for me.  I cannot see any reason why it would not work as there is very little to configure.

Webthings Version: Current version: 0.12.0
Add-on version: 0.3.2

**Comment by mrstegeman** [comment]
Does it work if you switch to speedtest.net in the add-on settings?

**Comment by da-wolff** [comment]
No unfortunately it doesn't.  I tried that first after finding the server ID I wanted & testing that with the speedtest-cli first.
I figured that fast.com may have been a simpler implementation to get working but neither seem to work for me.

**Comment by mrstegeman** [comment]
Maybe an obvious question, but does your gateway have internet connectivity?

**Comment by da-wolff** [comment]
I can perform the speedtest's by loading the appropriate webpage on my raspberry pi host.
I can also access my webthings dashboards via the remote mozilla address.
I can also update add-ons etc.


What does this indicate?
getValue for property download for: Internet speed test returning 0

**Comment by mrstegeman** [comment]
That's odd. Could you try playing around with those two modules manually? Maybe they'll dump out a useful error.

https://pypi.org/project/fastdotcom/
https://pypi.org/project/speedtest-cli/

**Comment by da-wolff** [comment]
> No unfortunately it doesn't. I tried that first after finding the server ID I wanted & testing that with the speedtest-cli first.
> I figured that fast.com may have been a simpler implementation to get working but neither seem to work for me.

As mentioned I had no problems running speedtest-cli & obtaining results.
FYI, worth mentioning that I am running my Webthings gateway as a docker container.  I do not have any problems with other addons performing as expected.  This includes some which are gathering data from weather sources online etc.

**Comment by mrstegeman** [comment]
Maybe related: https://github.com/sivel/speedtest-cli/issues/750

**Comment by da-wolff** [comment]
pi@raspberrypi:~ $ speedtest
Retrieving speedtest.net configuration...
Testing from TPG Internet (x.x.x.x)...
Retrieving speedtest.net server list...
Selecting best server based on ping...
Hosted by Telstra (Brisbane) [0.62 km]: 9.531 ms
Testing download speed................................................................................
Download: 46.48 Mbit/s
Testing upload speed................................................................................................
Upload: 14.64 Mbit/s
pi@raspberrypi:~ $

**Comment by mrstegeman** [comment]
Can you do that from within the Docker container, though?

**Comment by da-wolff** [comment]
OK, so inside Docker it was getting an error from speedtest-cli:
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get local issuer certificate 
I performed a certificate install process and that didn't fix it either.

Then I stumbled across this elsewhere & viola it then worked from speedtest-cli:

"sudo update-ca-certificates --fresh"

Following that re-enabling the add-on forced a new test and all good !!

Thanks for your assistance!  I had kind of assumed that whatever library functions required could be obtained outside of docker, but I am relatively new to the implementation of containers.
